Add checkout strategy to stash apply/pop methods

slavikus opened the pull request:
For a finer control over the unstash process, propogate checkout strategy to Objective-Git from the underlying libgit2 stash methods by adding an extra `strategy` argument to `-[GTRepository applyStashAtIndex:…]` and  `-[GTRepository popStashAtIndex:…]`

slavikus commented:
@tiennou , I'm going to flood you with these one by one now :D

pietbrauer commented:
@slavikus Could you rebase this branch onto master with #459 being merged?

slavikus commented:
@pietbrauer will do.

slavikus commented:
@pietbrauer there we go.

pietbrauer commented:
@slavikus So it's basically adding ` (with a default checkout strategy)` and two new lines or did something go wrong in the merging?

slavikus commented:
@pietbrauer In this PR, I've added checkout strategy as a separate parameter; in #459 , it was replaced by more generic (and more broad) GTCheckoutOptions class, so I've removed the more shallow variant that was previously used here.

pietbrauer commented:
Ok, so something went wrong. [There are only 4 changed lines](https://github.com/libgit2/objective-git/pull/606/files)

slavikus commented:
Nah, not really. #459 basically superceeded this one entirely. :)

slavikus commented:
However, now when I think of it, the  "(with a default checkout strategy)" makes no sense anymore. :) I'll push a clarifying commit.

pietbrauer commented:
Ok, cool. Thanks. No need to wait for CI then.
